plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiCommandLineProcessor.kt

```diff
@@ -60,6 +60,17 @@ class JvmAbiCommandLineProcessor : CommandLineProcessor {
                         "private top-level classes will no longer be available from Java classes in the same package.",
                 false,
             )
+        val REMOVE_INTERNAL_DECLARATIONS_OPTION: CliOption =
+            CliOption(
+                "removeInternalDeclarations",
+                "true/false",
+                "Remove internal declarations from from ABI. False by default due to backwards compatibility. If enabled, " +
+                        "internal functions are being removed and will no longer be available from Java if its being compiled against abi.jar. " +
+                        "Works best in conjunction with flags:" +
+                        "\n  removePrivateClasses - internal classes are being removed too;" +
```

Comment:
If that is the logic we're aiming for (internal classes are removed only if both options are enabled), then I guess the option should rather be named `treatInternalAsPrivate` (I think you even proposed it somewhere)? Because the current name would be a bit misleading.